### PR TITLE
update built binaries in CI

### LINKED ISSUE #1512: Releases should include no-relay but not rococo-local binary
Releases should include no-relay but not rococo-local binary.  However, both need to be built and the rococo-local used when comparing the metadata in the workflow.

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to update built binaries in CI and where they are being used. See issue description for more detail.

These are the binaries that will be posted in each GitHub release after this PR is merged:

Screenshot 2023-05-12 at 1 03 27 PM

Closes #1512

### CODE REVIEW

File: .github/workflows/verify-pr-commit.yml [hunk 1]
```diff
@@ -93,20 +74,20 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        network: [rococo, mainnet]
+        # network: [dev, local, rococo, mainnet]
```
Comment (wilwade) [inline .github/workflows/verify-pr-commit.yml:77]: @demisx Is this supposed to be commented out?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -865,13 +858,19 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        network: [local]
         arch: [amd64]
         node: [collator-node-local, instant-seal-node]
         include:
-          - network: local
+          - node: collator-node-local
```
Comment (wilwade) [inline .github/workflows/release.yml:864]: I'd be OK with not publishing this one at all now that I look at it. Fine to do so, but also fine to have less in the release.

**Review by wilwade — COMMENTED** [review]
One comment to check and an option to remove a release bin

File: .github/workflows/release.yml (hunk 2 above)
Comment (wilwade) [reply .github/workflows/release.yml:864]: Be nice if we could remove it from the build entirely if we don't use it elsewhere

**Review by saraswatpuneet — APPROVED** [review]

**Pull request merged**